app/main/service/auth_service.py

```diff
@@ -1 +1,206 @@
+
+        except Exception as e:
+            LOG.error('Login Failed')
+            LOG.debug(traceback.print_exc())
+            response_object = {
+                'status': 'fail',
+                'message': 'Try again',
+            }
+            return response_object, 500
+
+    @staticmethod
+    def logout_user(data):
+        try:
+            if not current_user.is_authenticated:
+                response_object = {
+                    'status': 'Invalid',
+                    'message': 'Not logged in',
+                }
+                return response_object, 300
+            logout_user()
+            response_object = {
+                'status': 'Success',
+                'message': 'Logged Out Successfully',
+            }
+            return response_object, 200
+        except Exception as e:
+            LOG.error('Logout Failed')
+            LOG.debug(traceback.print_exc())
+            response_object = {
+                'status': 'fail',
+                'message': 'Try again',
+            }
+            return response_object, 500
+
+    @staticmethod
+    def signup_user(data):
+        try:
+            user = User.query.filter_by(email=data.get('email')).first()
+            if user is not None:
+                response_object = {
+                    'status': 'Invalid',
+                    'message': 'User Already Exists',
+                }
+                LOG.info(
+                    'User already present in database. Redirect to Login Page')
+                return response_object, 300
+
+            user = User(data.get('id'), data.get('username'),
+                        data.get('password'), data.get('email'))
+            response_object = {
+                'status': 'Success',
+                'message': 'User added Successfully',
+            }
+            return response_object, 300
+
+        except Exception as e:
+            LOG.error('User with email {} couldn\'t be Signed Up. Please try again'.format(
+                data.get('email')))
+            LOG.debug(traceback.print_exc())
+            response_object = {
+                'status': 'fail',
+                'message': 'Try again',
+            }
+            return response_object, 500
+
+    @staticmethod
+    def send_verification(data):
+        try:
+            user = User.query.filter_by(email=data.get('email')).first()
+            if user.isVerified():
+                response_object = {
+                    'status': 'Invalid',
+                    'message': 'Email is already Verified.',
+                }
+                return response_object, 300
+
+            token = generate_confirmation_token(data.get('email'))
+            subject = "Hola! To hop onto IIT Tech Ambit, please confirm your email."
+            confirm_url = url_for('ConfirmToken.post',
+                                  token=token, _external=True)
+            async_send_mail(app, data.get('email'), subject, confirm_url)
```

Comment:
We don't send `app` here but rather `current_app._get_current_object()`. Also there is no import of `app` which won't be anyway since you've to import `current_app`